src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,165 @@
+                    hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithExpression(newInvocationExpr);
+
+
+                    //first iteration: adding AddJsonFile method
+
+                    //line 104: from builder.Configuration.ConfigureAppConfiguration() to builder.Configuration.AddJsonFile
+                    //line 105: from original invocation to builder.Configuration.AddJsonFile(new arguments)
+                    //line 106: from builder.Configuration.AddJsonFile to builder.Configuration.AddJsonFile(new arguments).AddJsonFile
+
+                    //second iteration: adding AddEnvironmentVariables method after AddJsonFile
+                    //line 104: from builder.Configuration.AddJsonFile(new arguments).AddJsonFile to builder.Configuration.AddJsonFile(new arguments).AddEnvironmentVariables
+                    //line 105: from builder.Configuration.AddJsonFile(new arguments) to builder.Configuration.AddJsonFile(new arguments).AddenvironmentVariables()
+                    //line 106: from builder.Configuration.AddJsonFile(new arguments).AddEnvironmentVariables to builder.Configuration.AddJsonFile(new arguments).AddEnvironmentVariables().AddEnvironmentVariables
+
+                    //so basically line 106 is just setting up for the next iteration, where the duplicated method is gonna be replaced with a new method
+                    //so the second AddJsonFile will be replaced with AddEnvironmentVariables
+                    //after the last iteration hostBasedInvocationMethodExpr will become an invocation with duplicated methods, which doesn't make any sense
+                    //but since the newInvocationExpr is already updated with the correct value, we're not gonna need hostBasedInvocationMethodExpr anymore
+                }
+            }
+
+            else if (initArgumentExpr is SimpleLambdaExpressionSyntax { } simpleLambdaExpr)
+            {
+                if (ExtractSimpleLambdaBody(simpleLambdaExpr) is not InvocationExpressionSyntax body)
+                {
+                    return editor.OriginalDocument;
+                }
+
+                var argument = body.ArgumentList;
+
+                if (body.Expression is not MemberAccessExpressionSyntax bodyExpression)
+                {
+                    return editor.OriginalDocument;
+                }
+
+                var method = bodyExpression.Name;
+
+                hostBasedInvocationMethodExpr = hostBasedInvocationMethodExpr.WithName(method);
+                newInvocationExpr = newInvocationExpr.WithExpression(hostBasedInvocationMethodExpr).WithArgumentList(argument);
+            }
+
+            editor.ReplaceNode(invocation, newInvocationExpr);
+        }
+
+        return editor.GetChangedDocument();
+    }
+    private static SyntaxList<StatementSyntax> ExtractParenLambdaBody(ParenthesizedLambdaExpressionSyntax initArgumentExpr)
+    {
+        var body = initArgumentExpr.Block.Statements;
```

Comment:
Do we need to do any validation on `Block` here? For example, check if it is null...